rules/Php85/Rector/FuncCall/ArrayKeyExistsNullToEmptyStringRector.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php85\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\BinaryOp\Coalesce;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Scalar\String_;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://wiki.php.net/rfc/deprecations_php_8_5#deprecate_using_values_null_as_an_array_offset_and_when_calling_array_key_exists
+ * @see \Rector\Tests\Php85\Rector\FuncCall\RemoveFinfoBufferContextArgRector\RemoveFinfoBufferContextArgRectorTest
+ */
+final class ArrayKeyExistsNullToEmptyStringRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace null key in array_key_exists with empty string',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+        array_key_exists($key, $array);
+        CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+        array_key_exists($key ?? '', $array);
+        CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+
+
+        if (! $node instanceof FuncCall) {
+            return null;
+        }
+
+        if (! $this->isName($node->name, 'array_key_exists')) {
+            return null;
+        }
+
+        $args = $node->getArgs();
+
+        if (count($args) < 2) {
+            return null;
+        }
+
+        $keyExpr = $args[0]->value;
+
+        if ($keyExpr instanceof Coalesce) {
+            return null;
+        }
+        
+
+        if ($keyExpr instanceof String_ && $keyExpr->value === '') {
+            return null;
+        }
+
+        if ($this->isName($keyExpr, 'null')) {
```

Comment:
use valueResolver->isNull instead, ensure exactly null variable, not variable named null